README.md

```diff
@@ -1,2 +1,91 @@
+
+* `template_filename`: *Optional.* *default `template_file.txt`* the file name for a generated template.
+
+## Behavior
+
+### `check`: Check for new messages that match the rules
+
+Check will report a new version `timestamp` if messages
+
+1: Are addressed to `@bot_name`.
+
+2: If `grammar` is defined it matches the regex of the message.
+
+#### `check Parameters`
+
+Check accepts no params.
+
+### `in`: Get the message
+
+A file **bender.json** will be created with the message payload.
+
+if `template` is defined it will be evaluated and written `template_filename`. For more info on syntax check [Template section](#Template) for example usage check the [examples](#Examples)
+
+#### `in Parameters`
+
+In accepts no params.
+
+### `out`: Reply to original message thread
+
+Sends a message as a thread to the original message sent by the user.
+
+#### `out Parameters`
+
+* `path`: *Required.* The path of the resource name. To fetch original message ts
+
+* `reply`: *Required.* The content of the reply. This can be also a template.
+
+### Template
+
+Under the hood the template use python [Jinj2](http://jinja.pocoo.org/docs/2.9/).
+
+* `variables`: You can access variables using curly braces `{{ VARIABLE_NAME }}`.
+   All environmental variables are accessible through `{{ ENV['PATH'] }}`. Concourse exposes some metadata info like job_name, build and ... for more info [check concourse website](https://concourse.ci/implementing-resources.html#resource-metadata)
+   The regex groups are accessible to the template engine as `{{ regex }}` if you used *subgroups* in your expression you can access each group with index `{{ regex[0] }}`. If you used *named subgroups* you can access them as dictionary `{{ regex['name'] }}`.
+
+* `white spaces`: You can use `\n` for new lines `\t`.
+
+* `filters`: You can use Jinja2 [built filters](http://jinja.pocoo.org/docs/2.9/templates/#builtin-filters).
+
+## Tips
+
+* Test your regex using something like [regexr.com](http://regexr.com/), It is better to use [named subgroups](http://www.regular-expressions.info/brackets.html).
+
+* Note that you need to escape `\` with another `\\` in your expression.
+
+* Use `\s+` between commands to give a little bit of room for user error on spaces.
+
+* You probably want to define **version: every** and in your **trigger: true** so the resource will go through all the messages and trigger the jobs.
+
+```yaml
+      - get: bender
+        version: every
+        trigger: true
+```
+
+* If you have multiple triggers on the same channel it is better to use a different bot token for each. check TODO
+
+## Examples
+
+Check [examples.md](examples.md)
+
+## TODO
+
+* At the moment we depend on `ts` to mark last message. So last read message is kept in slack rather then concourse native *check* version. The advantage of using slack it will be faster, but downside you can't have multiple triggers per channel with same token. this only effects the check method.
```

Comment:
this -> This, effects -> affects. `The advantage of using slack it will be faster` reads badly, maybe rephrase the sentence to smth like `The advantage of using slack is better speed`